### PR TITLE
Update DG: Add planned enhancements

### PR DESCRIPTION
**Pull request by alientian**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -726,6 +729,16 @@ Also, we understand that some users may not wish to type leading zeros for days
 ### **9. Allow Multiple Indexes Input for Delete Command**
 
 ### **10. Enhance Remark Feature**
+**Potential flaw in current implementation**
+Our app is unable to allow users to copy texts from the company detail panel in our UI.
+If users wish to add on to the existing remark, they need to re-type the existing remark into the command box then add in the new remark.
+This may not be a practical implementation, especially if the existing remark is long, which affects user experience negatively.
+
+**How the feature will be changed**
```
Comment (tayruxin) [inline docs/DeveloperGuide.md:737]: should be **Proposed Enhancement**

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -726,6 +729,16 @@ Also, we understand that some users may not wish to type leading zeros for days
...
+Edit the error message returned for `remark INDEX re/`.
+Currently, when `remark 1 re/` is entered, the error message returned is `Oops! Remark should not be empty. Please try again!`.
+This can be enhanced to return the existing remarks in the Message Box where the user can copy the content unlike in the company detail panel where the user is unable to do so.
+Hence, the `remark 1 re/` command is modified to become a way to copy and paste existing remarks so that remarks become cumulative.
```
Comment (tayruxin) [inline docs/DeveloperGuide.md:741]: Add in a section called **Examples** which include some sample output that the user will see or something

**Review by tayruxin — CHANGES_REQUESTED** [review]

**Review by tayruxin — APPROVED** [review]

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -812,6 +816,30 @@ The `DeleteCommandParser` will then split the string by commas and remove the co
 
 
 ### Enhance Remark Feature
+**Potential flaw in current implementation**
```
Comment (GlendaChong) [inline docs/DeveloperGuide.md:819]: Pascal case for headers

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -812,6 +816,30 @@ The `DeleteCommandParser` will then split the string by commas and remove the co
...
+Our app is unable to allow users to copy texts from the company detail panel in our UI.
+If users wish to add on to the existing remark, they need to re-type the existing remark into the command box then add in the new remark.
+This may not be a practical implementation, especially if the existing remark is long, which affects user experience negatively.
+
+**Proposed Enhancement**
+
+Edit the error message returned for `remark INDEX re/`.
+Currently, when `remark 1 re/` is entered, the error message returned is `Oops! Remark should not be empty. Please try again!`.
+This can be enhanced to return the existing remarks in the Message Box where the user can copy the content unlike in the company detail panel where the user is unable to do so.
+Hence, the `remark INDEX re/` command is modified to become a way to copy and paste existing remarks so that remarks become cumulative.
```
Comment (GlendaChong) [inline docs/DeveloperGuide.md:830]:
```suggestion
Hence, the `remark INDEX re/` command is modified such that if the user does not enter any remarks after `re/` prefix, the message displayed to the user will include their existing remarks. Should they wish to add on to their existing remarks, they can easily copy their existing remarks from the Message Box.
```

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -812,6 +816,30 @@ The `DeleteCommandParser` will then split the string by commas and remove the co
...
+
+**Examples**
+* Google is the first company in the list and the user wants to add remarks cumulatively.
+
+When `remark 1 re/` is entered, the Message Box will display as shown below.
```
Comment (GlendaChong) [inline docs/DeveloperGuide.md:835]:
```suggestion
When `remark 1 re/` is entered, the Message Box will be displayed as shown below.
```

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -812,6 +816,30 @@ The `DeleteCommandParser` will then split the string by commas and remove the co
...
+```
+Remarks in Google:
+Require experience in Java, Interview on 12/12/2023
+```
+The user can copy from the Message Box and add on his remark. A sample input is
```
Comment (GlendaChong) [inline docs/DeveloperGuide.md:840]:
```suggestion
The user can copy from the Message Box and add on to his remarks. A sample input of an updated remark will then be: 
```

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -812,6 +816,30 @@ The `DeleteCommandParser` will then split the string by commas and remove the co
...
+**Examples**
+* Google is the first company in the list and the user wants to add remarks cumulatively.
+
+When `remark 1 re/` is entered, the Message Box will display as shown below.
+```
+Remarks in Google:
+Require experience in Java, Interview on 12/12/2023
+```
+The user can copy from the Message Box and add on his remark. A sample input is 
+`remark 1 re/Require experience in Java, Interview on 12/12/2023, Interview went well!`
```
Comment (GlendaChong) [inline docs/DeveloperGuide.md:841]: will the success message remain the same? probably specify at the top

**Review by GlendaChong — CHANGES_REQUESTED** [review]

**Review by GlendaChong — APPROVED** [review]
lgtm

**Pull request merged**